pages/base.page.js

```diff
@@ -0,0 +1,18 @@
+const PAGE_TIMEOUT = 10000;
+class BasePage {
+
+async open(link) {
+    await browser.url(link);
+}
+
+async completeFields(element, text) {
+        await element.waitForClickable({ timeout: PAGE_TIMEOUT });
+        await element.setValue(text);
+    }
+
+async clickElement (element) {
+        await element.waitForClickable({ timeout: PAGE_TIMEOUT });
+        await element.click();
+    }
+}
+export default new BasePage();
```

Comment:
As this is an **abstract class**, it's meant to be used **indirectly** by inheriting from it, so you don't **instantiate** an object of type BasePage, you directly export the class such as:

```javascript
export default BasePage

```

This is different than with concrete classes like LoginPage or LogoutPage, that exports an instantiated object